d2layouts/d2near/layout.go

```diff
@@ -106,7 +170,34 @@ func WithoutConstantNears(ctx context.Context, g *d2graph.Graph) (nears []*d2gra
 			}
 		}
 	}
-	return nears
+	return constantNearGraphs
+}
+
+func pluckOutNearObjectAndEdges(g *d2graph.Graph, obj *d2graph.Object) (descendantsObjects []*d2graph.Object, edges []*d2graph.Edge) {
+	for i := 0; i < len(g.Edges); i++ {
+		edge := g.Edges[i]
+		if edge.Src == obj || edge.Dst == obj {
```

Comment:
i think this might double append. can you add this as a test. i think the edge would double append in this case

```
n: {
  near: top-center
  x
  inner: {
    y
  }
  x -> y
}
a; b; c
```

Reply:
ohh i see.